### PR TITLE
Add custom search with `filter` example

### PR DESCRIPTION
**Pull request by ginglis13**
#192 

Add section to README with example for updating index facets and running filtered search against updated index. Tested with movies example locally.

### CODE REVIEW

**Review by curquiza — DISMISSED** [review]
thanks @ginglis13 

bors merge


If you are participating in [Hacktoberfest](https://blog.meilisearch.com/contribute-hacktoberfest-2021/), and you would like to receive a small gift from MeiliSearch too, please complete [this form](https://meilisearch.typeform.com/to/rPexR5gq).

**Comment by ginglis13** [comment]
added sleep in to both code sections in `src/lib.rs` to make sure filterable attributes actually get set

File: src/lib.rs [hunk 1]
```diff
@@ -88,7 +88,120 @@
...
+//! #        Movie{id: 1, title: String::from("Carol"), genres: vec!["Romance".to_string(), "Drama".to_string()]},
+//! #        Movie{id: 2, title: String::from("Wonder Woman"), genres: vec!["Action".to_string(), "Adventure".to_string()]},
+//! #        Movie{id: 3, title: String::from("Life of Pi"), genres: vec!["Adventure".to_string(), "Drama".to_string()]},
+//! #        Movie{id: 4, title: String::from("Mad Max"), genres: vec!["Adventure".to_string(), "Science Fiction".to_string()]},
+//! #        Movie{id: 5, title: String::from("Moana"), genres: vec!["Fantasy".to_string(), "Action".to_string()]},
+//! #        Movie{id: 6, title: String::from("Philadelphia"), genres: vec!["Drama".to_string()]},
+//! #    ], Some("id")).await.unwrap();
+//! #    // Query movies (note that there is a typo)
+//! #    println!("{:?}", movies.search().with_query("carol").execute::<Movie>().await.unwrap().hits);
+//! # })}
```
Comment (curquiza) [inline src/lib.rs:134]:
```suggestion
```

**Review by curquiza — APPROVED** [review]
thanks

bors merge

**Pull request merged**